### PR TITLE
giada: 0.16.4 -> unstable-2021-09-24

### PR DESCRIPTION
**Pull request by 0x4A6F**
###### Motivation for this change
Upgrade to [0.18.2](https://github.com/monocasual/giada/releases/tag/v0.18.2) incorporating these [changes](https://github.com/monocasual/giada/compare/v0.16.4...v0.18.2).

- [ ] Waiting for a release with https://github.com/monocasual/giada/pull/509

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/giada/default.nix [hunk 1]
```diff
@@ -14,26 +16,38 @@
...
   pname = "giada";
-  version = "0.16.4";
+  version = "0.18.2";
 
   src = fetchFromGitHub {
     owner = "monocasual";
     repo = pname;
-    rev = "v${version}";
-    sha256 = "0qyx0bvivlvly0vj5nnnbiks22xh13sqlw4mfgplq2lbbpgisigp";
+    # Using https://github.com/monocasual/giada/pull/509 till a new release is done.
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/giada/default.nix:32]: Please clarify that we are using that commit on master, not the fork itself.

File: pkgs/applications/audio/giada/default.nix [hunk 2]
```diff
@@ -14,26 +16,38 @@
...
+, flac
+, libogg
+, libvorbis
+, libopus
 }:
 
 stdenv.mkDerivation rec {
   pname = "giada";
-  version = "0.16.4";
+  version = "0.18.2";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/giada/default.nix:27]: This needs to be in the unstable format.

File: pkgs/applications/audio/giada/default.nix [hunk 3]
```diff
@@ -14,26 +16,38 @@
...
 
   src = fetchFromGitHub {
     owner = "monocasual";
     repo = pname;
-    rev = "v${version}";
-    sha256 = "0qyx0bvivlvly0vj5nnnbiks22xh13sqlw4mfgplq2lbbpgisigp";
+    # Using https://github.com/monocasual/giada/pull/509 till a new release is done.
+    rev = "f117a8b8eef08d904ef1ab22c45f0e1fad6b8a56";
+    sha256 = "01hb981lrsyk870zs8xph5fm0z7bbffpkxgw04hq487r804mkx9j";
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/giada/default.nix:35]: Is nlohmann/json devendored?

File: pkgs/applications/audio/giada/default.nix [hunk 4]
```diff
@@ -50,11 +64,18 @@ stdenv.mkDerivation rec {
...
+    libvorbis
+    libopus
+    rtmidi
   ];
 
   postPatch = ''
     sed -i 's:"deps/json/single_include/nlohmann/json\.hpp":<nlohmann/json.hpp>:' \
         src/core/{conf,init,midiMapConf,patch}.cpp
+    sed -i 's:<RtMidi.h>:<${rtmidi.src}/RtMidi.h>:' \
+        src/{core/kernelMidi,gui/elems/config/tabMidi,utils/ver}.cpp
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/giada/default.nix:78]: Can we use substituteInPlace instead? Also please use / or | as a delimiter for sed.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by 0x4A6F** [comment]
Thanks, updated.

**Comment by tobiasBora** [comment]
Seems to solve https://github.com/NixOS/nixpkgs/issues/142178 thanks! To get input, you just need to enable it in the configuration file. I just have a strange "larsen" effect when I try to register audio, but I guess I first need to learn how to use the software ^^

**Pull request merged**